pkg/kubelet/kubelet_linux.go

```diff
@@ -49,3 +56,52 @@ func (kl *Kubelet) cgroupVersionCheck() error {
 	}
 	return nil
 }
+
+func (kl *Kubelet) startHealthCheck() error {
+	if runtime.GOOS != "linux" {
+		return fmt.Errorf("systemd watchdog can only be used on linux")
+	}
+
+	watchdogVal, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		return fmt.Errorf("could not check if systemd watchdog information: %v", err)
+	}
+	if watchdogVal == 0 {
+		return fmt.Errorf("no watchdog interval is configured")
+	}
+
+	// Periodically perform health checks and notify systemd.
+	kl.checkAndNotify(watchdogVal / 2)
+	return nil
+}
+
+func (kl *Kubelet) checkAndNotify(interval time.Duration) {
+	// The checks performed are the same as those for /healthz.
+	// see: server.InstallDefaultHandlers()
+	checkers := []healthz.HealthChecker{
+		healthz.PingHealthz,
+		healthz.LogHealthz,
+		healthz.NamedCheck("syncloop", kl.SyncLoopHealthCheck),
+	}
+
+	go wait.Forever(func() {
+		if err := doCheck(checkers); err != nil {
+			klog.ErrorS(err, "Stopping to notify watchdog")
+		} else {
+			ack, err := daemon.SdNotify(false, daemon.SdNotifyWatchdog)
+			if err != nil {
+				klog.ErrorS(err, "notify watchdog failed")
```

Comment:
let's retry it once if failed. We really need to avoid false positives here. So trying one extra time should be ok here